lib/model-builder.js

```diff
@@ -404,15 +405,8 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
 
     // Merging the settings
     var originalSubclassSettings = subClassSettings;
-
-    if (!subClassSettings.configurableModelMerge) {
-      // legacy behaviour applying a deep merge of parent and child model settings
-      subClassSettings = deepMerge(baseClassSettings, subClassSettings);
-    } else {
-      // configurable model merge behaviour if option is set
-      let mergePolicy = ModelClass.getMergePolicy();
-      subClassSettings = mergeSettings(baseClassSettings, subClassSettings, mergePolicy);
-    }
+    let mergePolicy = ModelClass.getMergePolicy(subClassSettings);
+    subClassSettings = mergeSettings(baseClassSettings, subClassSettings, mergePolicy);
```

Comment:
no more feature flag. The merge policy returned by `getMergePolicy()` is not influenced by the provided sub class setting `configureModelMerge`